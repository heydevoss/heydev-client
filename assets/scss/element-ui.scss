// Import Custom Variables
@import "variables";

/* icon font path, required */
$--font-path: '~element-ui/lib/theme-chalk/fonts';

@import "~element-ui/packages/theme-chalk/src/index";