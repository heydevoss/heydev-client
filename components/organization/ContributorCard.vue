<template>
  <div @click="redirectToUserPage">
    <el-row>
      <el-col>
        <el-card class="card" shadow="hover" :body-style="{ padding: '0px' }">
          <br /><img :src="contributor.avatarUrl" class="image" />
          <div style="padding: 0px;">
            <span>{{ contributor.login }}</span>

            <div class="description">
              <br /><span>{{ contributor.bio }}</span
              ><br />
              <br />
            </div>
          </div>
          <footer class="box-footer">
            <table>
              <tr>
                <td>{{ contributor.totalIssues }}</td>
                <td>{{ contributor.totalPullRequests }}</td>
                <td>{{ contributor.totalCommits }}</td>
              </tr>
              <tr>
                <td>Issues</td>
                <td>Pull Requests</td>
                <td>Commit</td>
              </tr>
            </table>
          </footer>
        </el-card>
      </el-col>
    </el-row>
  </div>
</template>

<script>
export default {
  name: 'ContributorCard',
  props: {
    contributor: Object
  },
  methods: {
    redirectToUserPage() {
      this.$router.replace(`/user/${this.contributor.login}`)
    }
  }
}
</script>

<style lang="scss" scoped>
table {
  border-collapse: separate;
  border-spacing: 13px;
  margin: 0 0 0 0;
}
.box-footer {
  position: absolute;
  align-self: center;
  bottom: 0;
}
.card {
  width: 250px;
  height: 350px;
  background-color: #47525e;
  color: white;
  cursor: pointer;
}

.description {
  text-align: center;
  font-size: 0.8rem;
  justify-content: center;
  margin: 3px;
}

.bottom {
  margin-top: 13px;
  line-height: 12px;
}

.button {
  padding: 0;
  float: right;
}

.image {
  border-radius: 100%;
  width: 100px;
  height: 100px;
  display: relative;
  position: center;
}
</style>
