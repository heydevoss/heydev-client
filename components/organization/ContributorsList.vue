<template>
  <div>
    <ApolloQuery
      :query="require('@/apollo/queries/getContributors.gql')"
      :variables="{ quantity }"
    >
      <template v-slot="{ result: { loading, error, data } }">
        <div v-if="loading" v-loading="loading" class="loading apollo" />

        <div v-else-if="error" class="error apollo">An error occured</div>

        <!-- Result -->
        <div v-else-if="data" class="contributors">
          <div
            v-for="contributor in data.organization.contributors"
            :key="contributor.login"
            class="contributor"
          >
            <ContributorCard :contributor="contributor" />
          </div>
        </div>
        <div v-else v-loading="true" class="loading" />
      </template>
    </ApolloQuery>
    <el-button class="button--show-more" @click="showMoreContributors"
      >Show more</el-button
    >
  </div>
</template>
<script>
import ContributorCard from '@/components/organization/ContributorCard'
export default {
  name: 'ContributorsList',
  components: {
    ContributorCard
  },
  data() {
    return {
      quantity: 6
    }
  },
  methods: {
    showMoreContributors() {
      this.quantity += 3
    }
  }
}
</script>
<style lang="scss" scoped>
.contributors {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.contributor {
  margin: 3px 3px;
}

.loading {
  display: flex;
  justify-content: center;
}

.button--show-more {
  margin-top: 10px;
}
</style>
