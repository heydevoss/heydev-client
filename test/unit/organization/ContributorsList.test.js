import Vue from 'vue'
import { mount } from '@vue/test-utils'
import ContributorsList from '@/components/organization/ContributorsList.vue'
import ContributorCard from '@/components/organization/ContributorCard.vue'
import { Button, Card } from 'element-ui'

Vue.use(Button)
Vue.use(Card)

describe('<ContributorsList/>', () => {
  const wrapper = mount(ContributorsList, {
    data: () => ({
      getRepositories: {
        organization: {
          contributors: [
            {
              login: 'user',
              name: 'User M',
              avatarUrl: '@/assets/avatar/default.ong',
              bio: 'Hey thats my test',
              websiteUrl: 'www.website.com',
              totalIssues: 12,
              totalCommits: 4,
              totalPullRequests: 3
            }
          ]
        }
      }
    }),
    stubs: {
      'el-card': Card,
      'el-button': Button,
      'contributor-card': ContributorCard,
      ApolloQuery: true
    }
  })

  it('should be a Vue instance and renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain button', () => {
    expect(wrapper.contains(Button)).toBe(true)
  })

  it('should increase quantity of contributors when button is clicked', () => {
    expect(wrapper.vm.quantity).toBe(6)
    const button = wrapper.find('button')
    button.trigger('click')
    expect(wrapper.vm.quantity).toBe(9)
  })
})
