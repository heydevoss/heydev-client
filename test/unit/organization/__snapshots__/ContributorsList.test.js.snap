// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContributorsList/> should be a Vue instance and renders correctly 1`] = `
<div>
  <apolloquery-stub
    query="[object Object]"
    variables="[object Object]"
  />
   
  <button
    class="el-button button--show-more el-button--default"
    type="button"
  >
    <!---->
    <!---->
    <span>
      Show more
    </span>
  </button>
</div>
`;
